src/@next/components/organisms/ProductVariantPicker/ProductVariantAttributeSelect.tsx

```diff
@@ -0,0 +1,149 @@
+import React from "react";
+
+import { Icon, Input } from "@components/atoms";
+import { InputSelect } from "@components/molecules";
+import { ProductDetails_product_variants } from "@sdk/queries/types/ProductDetails";
+import {
+  IProductVariantsAttribute,
+  IProductVariantsAttributesSelectedValues,
+} from "@types";
+
+import { useSelectableProductVariantsAttributeValues } from "../../../hooks/useSelectableProductVariantsAttributeValues";
+import { SelectSidebar } from "../SelectSidebar";
+import * as S from "./styles";
+
+export const ProductVariantAttributeSelect: React.FC<{
+  selectSidebar: boolean;
+  selectSidebarTarget?: HTMLElement | null;
+  productVariantsAttributeId: string;
+  productVariants: ProductDetails_product_variants[];
+  productVariantsAttribute: IProductVariantsAttribute;
+  productVariantsAttributesSelectedValues: IProductVariantsAttributesSelectedValues;
+  onChangeSelection: (value: any, name?: any) => void;
+  onClearSelection: () => void;
+}> = ({
+  selectSidebar = false,
+  selectSidebarTarget,
+  productVariantsAttributeId,
+  productVariants,
+  productVariantsAttribute,
+  productVariantsAttributesSelectedValues,
+  onChangeSelection,
+  onClearSelection,
+}) => {
+  const [showSelectSidebar, setShowSelectSidebar] = React.useState(false);
+
+  const selectableProductVariantsAttributeValues = useSelectableProductVariantsAttributeValues(
+    productVariantsAttributeId,
+    productVariants,
+    productVariantsAttributesSelectedValues
+  );
+
+  const selectedValue =
+    productVariantsAttributesSelectedValues &&
+    productVariantsAttributesSelectedValues[productVariantsAttributeId]
+      ? {
+          disabled: false,
+          id: productVariantsAttributesSelectedValues[
+            productVariantsAttributeId
+          ]!.id,
+          label: productVariantsAttributesSelectedValues[
+            productVariantsAttributeId
+          ]!.name!,
+          value: productVariantsAttributesSelectedValues[
+            productVariantsAttributeId
+          ]!.value!,
+        }
+      : null;
+
+  const attributeOptions = productVariantsAttribute.values
+    .filter(value => value)
+    .map(value => {
+      const isOptionDisabled =
+        selectableProductVariantsAttributeValues[productVariantsAttributeId] &&
+        !selectableProductVariantsAttributeValues[
+          productVariantsAttributeId
+        ].values.includes(value);
+
+      return {
+        disabled: isOptionDisabled,
+        id: value.id,
+        label: value.name!,
+        value: value.value!,
+      };
+    });
+
+  const selectLabel = productVariantsAttribute.attribute.name
+    ? productVariantsAttribute.attribute.name!
+    : "";
+
+  const selectedValuesList = selectedValue ? [selectedValue.value] : [];
+
+  const disabledValuesList = attributeOptions
+    .filter(optionValue => optionValue.disabled)
+    .map(optionValue => optionValue.value);
+
+  const onSelectValueInSidebar = (optionValue: string) => {
```

Comment:
Functions that get called **on**Something are handlers, consider renaming to `handleSelectValueInSidebar`
The general convention is:
`<C onClick={handleClick} />`